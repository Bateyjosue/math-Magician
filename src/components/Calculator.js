import React from 'react';
import calculate from '../logic/calculate';

export default class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: null,
      next: null,
      operation: null,
    };
  }

  handleClick = (e) => {
    const data = e.target.textContent;
    const calculation = calculate(this.state, data);
    this.setState(calculation);
  }

  render() {
    const { total, next, operation } = this.state;
    return (
      <div className="calc">
        <div className="screen">
          { next || operation || total || 0}
        </div>
        <div className="operation">
          <div>
            <button type="button" onClick={this.handleClick} value="&divide;">รท</button>
            <button type="button" onClick={this.handleClick} value="x">x</button>
            <button type="button" onClick={this.handleClick} value="-">-</button>
            <button type="button" onClick={this.handleClick} value="+">+</button>
            <button type="button" onClick={this.handleClick} value="=">=</button>
          </div>
        </div>
        <div className="operand">
          <div>
            <button type="button" onClick={this.handleClick} value="AC">AC</button>
            <button type="button" onClick={this.handleClick} value="&plusmn;">+/-</button>
            <button type="button" onClick={this.handleClick} value="%">%</button>
            <button type="button" onClick={this.handleClick} value="7">7</button>
            <button type="button" onClick={this.handleClick} value="8">8</button>
            <button type="button" onClick={this.handleClick} value="9">9</button>
            <button type="button" onClick={this.handleClick} value="4">4</button>
            <button type="button" onClick={this.handleClick} value="5">5</button>
            <button type="button" onClick={this.handleClick} value="6">6</button>
            <button type="button" onClick={this.handleClick} value="1">1</button>
            <button type="button" onClick={this.handleClick} value="2">2</button>
            <button type="button" onClick={this.handleClick} value="3">3</button>
            <button type="button" onClick={this.handleClick} value="0">0</button>
            <button type="button"> </button>
            <button type="button" onClick={this.handleClick} value=".">.</button>
          </div>
        </div>
      </div>
    );
  }
}
