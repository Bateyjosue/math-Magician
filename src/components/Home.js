import React from 'react';
import stylus from './styles/Home.module.css';

export default function Home() {
  return (
    <main>
      <h4 className={stylus.h4}>Welcome to our Page!</h4>
      <p className={stylus.pa}>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Saepe, quisquam,
        natus, odit voluptatibus commodi mollitia beatae iste dignissimos distinctio
        enim quidem quasi quae accusantium! Necessitatibus, nam. Cupiditate earum
        deserunt quo ea consectetur exercitationem aliquid quae. Explicabo quae
        veniam minus earum soluta repellendus qui sit eligendi maxime fugit
        perferendis ullam, ipsa dignissimos rerum tempore minima? Vero, asperiores
        fugiat nulla ullam tempore quod perspiciatis molestiae quas iusto excepturi
        ratione sunt, optio quos. Adipisci facilis delectus ducimus voluptatum itaque
        sequi minima non nihil repudiandae cum officia consectetur odit ratione quod
        expedita totam quo, tenetur vel minus quam! Esse, nisi odio! Ullam, enim est!
      </p>
    </main>
  );
}
