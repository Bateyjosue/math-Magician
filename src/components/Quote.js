import React from 'react';
import style from './styles/Quote.module.css';

export default function Quote() {
  return (
    <div>
      <p className={style.par}>
        Mathematics is not about numbers, equations, computations, or
        algorithms; it is about understanding. __ William Paul Thurston
      </p>
    </div>
  );
}
