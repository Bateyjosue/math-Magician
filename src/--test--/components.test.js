import { render } from '@testing-library/react';
import Home from '../components/Home';
import Quote from '../components/Quote';
import Calculator from '../components/Calculator';

describe('Home Component Test', () => {
  test('should render Home DOM', () => {
    const homeTree = render(<Home />);
    expect(homeTree).toMatchSnapshot();
  });
});

describe('Quote Component Test', () => {
  test('should render Quote on the DOM', () => {
    const quoteTree = render(<Quote />);
    expect(quoteTree).toMatchSnapshot();
  });
});

describe('Calculator Component Test', () => {
  test('should render Calculation', () => {
    const calculatorTree = render(<Calculator />);
    expect(calculatorTree).toMatchSnapshot();
  });
});
