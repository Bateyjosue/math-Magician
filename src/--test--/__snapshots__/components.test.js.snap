// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Component Test should render Calculation 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <h4>
          Lets do some math!
        </h4>
        <div
          class="calc"
        >
          <div
            class="screen"
          >
            0
          </div>
          <div
            class="operation"
          >
            <div>
              <button
                type="button"
                value="÷"
              >
                ÷
              </button>
              <button
                type="button"
                value="x"
              >
                x
              </button>
              <button
                type="button"
                value="-"
              >
                -
              </button>
              <button
                type="button"
                value="+"
              >
                +
              </button>
              <button
                type="button"
                value="="
              >
                =
              </button>
            </div>
          </div>
          <div
            class="operand"
          >
            <div>
              <button
                type="button"
                value="AC"
              >
                AC
              </button>
              <button
                type="button"
                value="±"
              >
                +/-
              </button>
              <button
                type="button"
                value="%"
              >
                %
              </button>
              <button
                type="button"
                value="7"
              >
                7
              </button>
              <button
                type="button"
                value="8"
              >
                8
              </button>
              <button
                type="button"
                value="9"
              >
                9
              </button>
              <button
                type="button"
                value="4"
              >
                4
              </button>
              <button
                type="button"
                value="5"
              >
                5
              </button>
              <button
                type="button"
                value="6"
              >
                6
              </button>
              <button
                type="button"
                value="1"
              >
                1
              </button>
              <button
                type="button"
                value="2"
              >
                2
              </button>
              <button
                type="button"
                value="3"
              >
                3
              </button>
              <button
                type="button"
                value="0"
              >
                0
              </button>
              <button
                type="button"
              >
                 
              </button>
              <button
                type="button"
                value="."
              >
                .
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <h4>
        Lets do some math!
      </h4>
      <div
        class="calc"
      >
        <div
          class="screen"
        >
          0
        </div>
        <div
          class="operation"
        >
          <div>
            <button
              type="button"
              value="÷"
            >
              ÷
            </button>
            <button
              type="button"
              value="x"
            >
              x
            </button>
            <button
              type="button"
              value="-"
            >
              -
            </button>
            <button
              type="button"
              value="+"
            >
              +
            </button>
            <button
              type="button"
              value="="
            >
              =
            </button>
          </div>
        </div>
        <div
          class="operand"
        >
          <div>
            <button
              type="button"
              value="AC"
            >
              AC
            </button>
            <button
              type="button"
              value="±"
            >
              +/-
            </button>
            <button
              type="button"
              value="%"
            >
              %
            </button>
            <button
              type="button"
              value="7"
            >
              7
            </button>
            <button
              type="button"
              value="8"
            >
              8
            </button>
            <button
              type="button"
              value="9"
            >
              9
            </button>
            <button
              type="button"
              value="4"
            >
              4
            </button>
            <button
              type="button"
              value="5"
            >
              5
            </button>
            <button
              type="button"
              value="6"
            >
              6
            </button>
            <button
              type="button"
              value="1"
            >
              1
            </button>
            <button
              type="button"
              value="2"
            >
              2
            </button>
            <button
              type="button"
              value="3"
            >
              3
            </button>
            <button
              type="button"
              value="0"
            >
              0
            </button>
            <button
              type="button"
            >
               
            </button>
            <button
              type="button"
              value="."
            >
              .
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Home Component Test should render Home DOM 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main>
        <h4
          class="h4"
        >
          Welcome to our Page!
        </h4>
        <p
          class="pa"
        >
          Mathematics as an expression of the human mind reflects the active will, the contemplative reason, and the desire for aesthetic perfection. Its basic elements are logic and intuition, analysis and construction, generality and individuality.
        </p>
        <p
          class="pa"
        >
          One of the most amazing things about mathematics is the people who domath aren't usually interested in application, because mathematics itself is truly a beautiful art form. It's structures and patterns,and that's what we love, and that's what we get off on.
        </p>
      </main>
    </div>
  </body>,
  "container": <div>
    <main>
      <h4
        class="h4"
      >
        Welcome to our Page!
      </h4>
      <p
        class="pa"
      >
        Mathematics as an expression of the human mind reflects the active will, the contemplative reason, and the desire for aesthetic perfection. Its basic elements are logic and intuition, analysis and construction, generality and individuality.
      </p>
      <p
        class="pa"
      >
        One of the most amazing things about mathematics is the people who domath aren't usually interested in application, because mathematics itself is truly a beautiful art form. It's structures and patterns,and that's what we love, and that's what we get off on.
      </p>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Quote Component Test should render Quote on the DOM 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="maincontainer"
      >
        <p
          class="par"
        >
          Mathematics is not about 
          <span
            style="background: rgb(218, 51, 17);"
          >
            numbers
          </span>
          ,
           
          <span
            style="background: rgb(218, 51, 17);"
          >
            equations
          </span>
          ,
           
          <span
            style="background: rgb(218, 51, 17);"
          >
            computations
          </span>
           , or 
          <span
            style="background: rgb(218, 51, 17);"
          >
            algorithms
          </span>
          ; it is about understanding. __ William Paul Thurston
        </p>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="maincontainer"
    >
      <p
        class="par"
      >
        Mathematics is not about 
        <span
          style="background: rgb(218, 51, 17);"
        >
          numbers
        </span>
        ,
         
        <span
          style="background: rgb(218, 51, 17);"
        >
          equations
        </span>
        ,
         
        <span
          style="background: rgb(218, 51, 17);"
        >
          computations
        </span>
         , or 
        <span
          style="background: rgb(218, 51, 17);"
        >
          algorithms
        </span>
        ; it is about understanding. __ William Paul Thurston
      </p>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
